### PR TITLE
Adds JavaFX Testing to Build

### PR DESCRIPTION
**Pull request by JLLeitschuh**
JavaFX tests should now run as part of the build.
Tests should work headless on Travis.

### CODE REVIEW

File: src/main/java/edu/wpi/grip/core/Operation.java [hunk 1]
```diff
@@ -22,6 +22,10 @@
      */
     String getDescription();
 
+    default String getCSSName(){
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/Operation.java]: Operation should not have anything to do with CSS, since it's not in the gui package.

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/Operation.java]: Where should this method reside then?

**Comment by JLLeitschuh** [comment]
22%! Jeez! That's much higher than I expected.

File: src/test/java/edu/wpi/grip/ui/controllers/MainWindowControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import static org.testfx.api.FxAssert.verifyThat;
+
+
+public class MainWindowControllerTest extends ApplicationTest {
+
+    @Override
+    public void start(Stage stage) throws Exception {
+        new Main().start(stage);
+        sleep(5, TimeUnit.SECONDS);
```
Comment (ThomasJClark) [inline src/test/java/edu/wpi/grip/ui/controllers/MainWindowControllerTest.java]: Why sleep for 5 seconds?

File: src/main/java/edu/wpi/grip/core/Operation.java (hunk 1 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/Operation.java]: Or I could rename it to `getHyphenatedName`

File: src/test/java/edu/wpi/grip/ui/controllers/MainWindowControllerTest.java (hunk 2 above)
Comment (JLLeitschuh) [inline src/test/java/edu/wpi/grip/ui/controllers/MainWindowControllerTest.java]: - [ ] Ooops, left over from testing. Will remove.

File: src/main/java/edu/wpi/grip/core/Operation.java (hunk 1 above)
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/Operation.java]: I would not put this method in `Operation` at all.

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/Operation.java]: I need the method somewhere. Where would it be best located?

File: src/main/java/edu/wpi/grip/core/Operation.java [hunk 3]
```diff
@@ -22,6 +22,10 @@
      */
     String getDescription();
 
+    default String getHyphenatedName(){
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/Operation.java]: If you want this somewhere else please offer a proposition as to where it should reside.

Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/Operation.java]: `StepView` would make sense since it's the only place that uses this to set a CSS class.

It looks like you also use the same value to set the ID in `OperationView`, so you'll also need some way to compute that.  I don't see any reason why this needs to be the same value as a CSS class used in `StepView`, so this can be a separate thing.

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/Operation.java]: I mean in all practicality the name and the description are purely UI elements as well. The are never actually used inside of the pipeline.

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package edu.wpi.grip.ui.util;
+
+import edu.wpi.grip.core.Connection;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.Step;
+
+public final class CSSElementUtility {
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: I like that this isn't in a random default method in the core package anymore, but there has to be a better name for this class.  Something that makes it clear that this is for looking up elements in unit tests.  Also documentation that makes that clear.

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    private CSSElementUtility(){}
+
+
+    /**
+     * Returns the CSS id name for an operation
+     * @param operation The operation to get the class name for
+     * @return The CSS id for the operation. To use as a css selector then prepend the string with a '#'
+     */
+    public static String idNameFor(Operation operation){
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: What does this have to do with CSS?

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    private static StringBuilder nameFor(Operation operation){
+        return shortNameFor(operation).append("-operation");
+    }
+
+    /**
+     * Creates the base short name for the operation.
+     * @param operation The operation.
+     * @return The short base of the operation's name.
+     */
+    private static StringBuilder shortNameFor(Operation operation){
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: The documentation says literally nothing about this method - it's just a rephrasing of the method name and signature.

I don't see any use for it, and since there's no justification in the documentation, I think it should probably not exist.

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                + "-to-" +
+                connection.getInputSocket().getSocketHint().getIdentifier();
+    }
+
+    /**
+     * Creates the base name for the operation.
+     * @param operation The operation to generate the name for.
+     * @return The base of the operation's name.
+     */
+    private static StringBuilder nameFor(Operation operation){
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: Why obfuscate this by splitting up into a bunch of different methods that pass around a `StringBuilder`?  I would just write the above methods using string concatenation.

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java (hunk 5 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: ID vs Class
The id is assigned as the elements ID param instead of assigning it as a class on the element.

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java (hunk 7 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: I had another method that used it before but removed it.

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java (hunk 4 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: Do you have a suggested name?

File: src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java (hunk 6 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java]: Its private so it doesn't really matter anyways. I'll remove it.
